cmake/list_append_keys_from.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
+#
+macro(list_append_keys_from listname source)
```

Comment:
I would suggest passing an optional keyword argument like `CONTEXT` for this purpose.